src/SickSafetyscannersRos.cpp

```diff
@@ -382,7 +382,20 @@ SickSafetyscannersRos::createLaserScanMessage(const sick::datastructure::Data& d
 {
   sensor_msgs::LaserScan scan;
   scan.header.frame_id = m_frame_id;
-  scan.header.stamp    = ros::Time::now();
+
+  // ros::Time ros_time = ros::Time::now();
+  // ROS_INFO_STREAM_THROTTLE(2, "ros time: sec " << ros_time.sec << " nsec " << ros_time.nsec);
+
+  std::shared_ptr<sick::datastructure::DataHeader> data_header = data.getDataHeaderPtr();
+  // ROS_INFO_STREAM_THROTTLE(2, "header date: " << data_header->getTimestampDate());
+  // ROS_INFO_STREAM_THROTTLE(2, "header time: " << data_header->getTimestampTime());
+
+  double data_ts = (double)(data_header->getTimestampDate() /*days since 01/01/1972*/
```

Comment:
Why C style casting? Also, this is interesting. Sick's data structure is giving `TimestampDate` from 1972 rather than 1970 :thinking:

Reply:
Main reason is that it is shorter than using `static_cast<double>`.

Reply:
I would still use the cpp style casting :)

Reply:
Casting removed. The compiler would implicitly convert them.